exporters/trace_stackdriver/src/main/java/io/opencensus/trace/exporter/StackdriverV1ExporterHandler.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace.exporter;
+
+import static com.google.api.client.util.Preconditions.checkNotNull;
+
+import com.google.auth.Credentials;
+import com.google.common.io.BaseEncoding;
+import com.google.devtools.cloudtrace.v1.PatchTracesRequest;
+import com.google.devtools.cloudtrace.v1.Trace;
+import com.google.devtools.cloudtrace.v1.TraceServiceGrpc;
+import com.google.devtools.cloudtrace.v1.TraceServiceGrpc.TraceServiceBlockingStub;
+import com.google.devtools.cloudtrace.v1.TraceSpan;
+import com.google.devtools.cloudtrace.v1.Traces;
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.auth.MoreCallCredentials;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.common.Timestamp;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanExporter.Handler;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A class that contains helper methods to convert {@link SpanData} to Stackdriver Trace API v1
+ * trace messages.
+ */
+final class StackdriverV1ExporterHandler extends Handler {
+  private static final String API_HOST = "cloudtrace.googleapis.com";
+  private static final String STATUS_CODE = "g.co/status/code";
+  private static final String STATUS_DESCRIPTION = "g.co/status/description";
+
+  private final String projectId;
+  private final TraceServiceBlockingStub traceServiceBlockingStub;
+
+  StackdriverV1ExporterHandler(Credentials credentials, String projectId) {
+    checkNotNull(credentials, "credentials");
+    this.projectId = checkNotNull(projectId, "projectId");
+    Channel channel = ManagedChannelBuilder.forTarget(API_HOST).build();
+    traceServiceBlockingStub =
+        TraceServiceGrpc.newBlockingStub(channel)
+            .withCallCredentials(MoreCallCredentials.from(credentials));
+  }
+
+  Trace generateTrace(SpanData spanData) {
```

Comment:
This method should have some tests.

Reply:
I will add tests in the next PR. I want to merge this and release a "-SNAPSHOT" version which allows other to start playing with the library. Before the 0.6.0 we should have tests I agree.

Reply:
Could you please add a TODO?